### DISCUSSION TITLE
Extra metadata linked to chunk when streaming

### DISCUSSION THREAD

**Discussion by antoineandrieu** [deepgram/community#609] [General help] [answered]
How can I incorporate and handle additional metadata specific to each audio chunk during live transcription processes, using the Python SDK? This metadata is unique to each audio chunk rather than associated with the client. Are there established methods or best practices within Python SDKs for transmitting and receiving this kind of chunk-specific metadata alongside the audio data for live transcription?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
Hi @antoineandrieu 

I dont know if there are any best practices for doing this (someone else can chime in with this if there is), but you would minimally need the following:
- place to store the metadata
  - to my knowledge, we don't provide anything like that. this is a datastore of some sort.
- an association of the metadata to the data
  - this would need to be something like a tuple of [transcription id, the type of data {utterance, paragraph, word}, identifier for the data (maybe start time), unique metadata id (could just be an INT or GUID)]
- the metadata itself with the ID (INT or GUID)

There are things that might naturally lend itself to this implementation, like a graph database.
